// @vkontakte/vkui-tokens/build/compilers/cssVars/declarations/compileBreakpointsCssVarsDeclaration.test.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var globals_1 = require("@jest/globals");
var common_tags_1 = require("common-tags");
var compileBreakpointsCssVarsDeclaration_1 = require("./compileBreakpointsCssVarsDeclaration");
(0, globals_1.describe)('compileBreakpointsCssVarsDeclaration', function () {
    (0, globals_1.it)('should not work without breakpoints', function () {
        var theme = {};
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe(null);
    });
    (0, globals_1.it)('should work with breakpoints and empty vals', function () {
        var theme = {
            breakpoints: {
                touch: {
                    breakpoint: 0,
                    adaptiveValue: 'compactX',
                },
                desktopS: {
                    breakpoint: 768,
                    adaptiveValue: 'regular',
                },
            },
        };
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe(null);
    });
    (0, globals_1.it)('should work with non adaptive values', function () {
        var theme = {
            breakpoints: {
                touch: {
                    breakpoint: 0,
                    adaptiveValue: 'compactX',
                },
                desktopS: {
                    breakpoint: 768,
                    adaptiveValue: 'regular',
                },
            },
            x1: {
                name: '--vkui--x1',
                value: 'var(--vkui--x1, 4px)',
                originalValue: '4px',
            },
        };
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe(null);
    });
    (0, globals_1.it)('should work with flat adaptive values and several breakpotins', function () {
        var theme = {
            breakpoints: {
                touch: {
                    breakpoint: 0,
                    adaptiveValue: 'compact',
                },
                desktopS: {
                    breakpoint: 768,
                    adaptiveValue: 'regular',
                },
            },
            sizePopupBasePadding: {
                auto: {
                    name: '--vkui--size_popup_base_padding',
                    value: 'var(--vkui--size_popup_base_padding)',
                },
                compact: {
                    name: '--vkui--size_popup_base_padding--compact',
                    value: 'var(--vkui--size_popup_base_padding--compact, 16px)',
                    originalValue: '16px',
                },
                regular: {
                    name: '--vkui--size_popup_base_padding--regular',
                    value: 'var(--vkui--size_popup_base_padding--regular, 20px)',
                    originalValue: '20px',
                },
            },
        };
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe((0, common_tags_1.stripIndent)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t\t\t:root {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t}\n\n\t\t\t@media (max-width: 767px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--compact);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-touch {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--compact);\n\t\t\t}\n\n\t\t\t@media (min-width: 768px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-desktop_s {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t}\n\t\t\t"], ["\n\t\t\t:root {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t}\n\n\t\t\t@media (max-width: 767px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--compact);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-touch {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--compact);\n\t\t\t}\n\n\t\t\t@media (min-width: 768px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-desktop_s {\n\t\t\t\t--vkui--size_popup_base_padding: var(--vkui--size_popup_base_padding--regular);\n\t\t\t}\n\t\t\t"]))));
    });
    (0, globals_1.it)('should work with flat adaptive values and several breakpotins and without overriding', function () {
        var theme = {
            breakpoints: {
                touch: {
                    breakpoint: 0,
                    adaptiveValue: 'compact',
                },
                desktopS: {
                    breakpoint: 768,
                    adaptiveValue: 'regular',
                },
            },
            sizePopupBasePadding: {
                auto: {
                    name: '--vkui--size_popup_base_padding',
                    value: 'var(--vkui--size_popup_base_padding)',
                },
                regular: {
                    name: '--vkui--size_popup_base_padding--regular',
                    value: 'var(--vkui--size_popup_base_padding--regular, 20px)',
                    originalValue: '20px',
                },
            },
        };
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe((0, common_tags_1.stripIndent)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\t\t:root {\n\t\t\t\t--vkui--size_popup_base_padding: 20px;\n\t\t\t}\n\t\t\t"], ["\n\t\t\t:root {\n\t\t\t\t--vkui--size_popup_base_padding: 20px;\n\t\t\t}\n\t\t\t"]))));
    });
    (0, globals_1.it)('should work with adaptive complex object', function () {
        var theme = {
            breakpoints: {
                touch: {
                    breakpoint: 0,
                    adaptiveValue: 'compact',
                },
                desktopS: {
                    breakpoint: 768,
                    adaptiveValue: 'regular',
                },
            },
            fontH1: {
                auto: {
                    fontSize: {
                        name: '--vkui--font_h1--font_size',
                        value: 'var(--vkui--font_h1--font_size)',
                    },
                    lineHeight: {
                        name: '--vkui--font_h1--line_height',
                        value: 'var(--vkui--font_h1--line_height)',
                    },
                },
                compact: {
                    fontSize: {
                        name: '--vkui--font_h1--font_size--compact',
                        value: 'var(--vkui--font_h1--font_size--compact, 24px)',
                        originalValue: '24px',
                    },
                },
                regular: {
                    fontSize: {
                        name: '--vkui--font_h1--font_size--regular',
                        value: 'var(--vkui--font_h1--font_size--regular, 22px)',
                        originalValue: '22px',
                    },
                    lineHeight: {
                        name: '--vkui--font_h1--line_height--regular',
                        value: 'var(--vkui--font_h1--line_height--regular, 30px)',
                        originalValue: '30px',
                    },
                },
            },
        };
        (0, globals_1.expect)((0, compileBreakpointsCssVarsDeclaration_1.compileBreakpointsCssVarsDeclaration)(theme)).toBe((0, common_tags_1.stripIndent)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\t\t\t:root {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t\t--vkui--font_h1--line_height: var(--vkui--font_h1--line_height--regular);\n\t\t\t}\n\n\t\t\t@media (max-width: 767px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--compact);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-touch {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--compact);\n\t\t\t}\n\n\t\t\t@media (min-width: 768px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-desktop_s {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t}\n\t\t\t"], ["\n\t\t\t:root {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t\t--vkui--font_h1--line_height: var(--vkui--font_h1--line_height--regular);\n\t\t\t}\n\n\t\t\t@media (max-width: 767px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--compact);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-touch {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--compact);\n\t\t\t}\n\n\t\t\t@media (min-width: 768px) {\n\t\t\t\t:root, .vkui--force-auto {\n\t\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t\t}\n\t\t\t}\n\n\t\t\t.vkui--force-desktop_s {\n\t\t\t\t--vkui--font_h1--font_size: var(--vkui--font_h1--font_size--regular);\n\t\t\t}\n\t\t\t"]))));
    });
});
var templateObject_1, templateObject_2, templateObject_3;
