// @vkontakte/vkui-tokens/utils/common/fontSizes/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fontSizes = exports.fontWeightAccentBold = exports.fontWeightAccent = exports.fontWeightBold = exports.fontWeight = exports.fontFamilyAccent = exports.fontFamily = void 0;
var helpers_1 = require("../helpers");
var sizeBase = helpers_1.helpers.sizeBase;
// базовые параметры шрифта
exports.fontFamily = 'Helvetica, Arial, sans-serif';
exports.fontFamilyAccent = 'MailSans, Helvetica, Arial, sans-serif';
exports.fontWeight = 400;
exports.fontWeightBold = 700;
exports.fontWeightAccent = 400;
exports.fontWeightAccentBold = 500;
// шрифтовая шкала
var fontSize1 = {
    fontSize: 11,
    lineHeight: "".concat(sizeBase * 3, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize2 = {
    fontSize: 13,
    lineHeight: "".concat(sizeBase * 5, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize3 = {
    fontSize: 15,
    lineHeight: "".concat(sizeBase * 5, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize4 = {
    fontSize: 17,
    lineHeight: "".concat(sizeBase * 6, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize5 = {
    fontSize: 20,
    lineHeight: "".concat(sizeBase * 7, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize6 = {
    fontSize: 24,
    lineHeight: "".concat(sizeBase * 8, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize7 = {
    fontSize: 32,
    lineHeight: "".concat(sizeBase * 10, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
var fontSize8 = {
    fontSize: 44,
    lineHeight: "".concat(sizeBase * 13, "px"),
    fontWeight: exports.fontWeight,
    fontFamily: exports.fontFamily,
};
exports.fontSizes = {
    fontSize1: fontSize1,
    fontSize2: fontSize2,
    fontSize3: fontSize3,
    fontSize4: fontSize4,
    fontSize5: fontSize5,
    fontSize6: fontSize6,
    fontSize7: fontSize7,
    fontSize8: fontSize8,
};
