// @vkontakte/vkui-tokens/build/compilers/cssVars/jsUtils/compileGetDeclarationString.test.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var globals_1 = require("@jest/globals");
var common_tags_1 = require("common-tags");
var compileGetDeclarationString_1 = require("./compileGetDeclarationString");
(0, globals_1.describe)('compileGetDeclarationString', function () {
    (0, globals_1.it)('should work with empty', function () {
        var theme = {};
        (0, globals_1.expect)((0, compileGetDeclarationString_1.compileGetDeclarationString)(theme)).toBe((0, common_tags_1.stripIndent)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarString(): string {\n\t\t\t\treturn ``;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarObject(): Properties {\n\t\t\t\treturn {} as Properties;\n\t\t\t}\n\t\t\t"], ["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarString(): string {\n\t\t\t\treturn \\`\\`;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarObject(): Properties {\n\t\t\t\treturn {} as Properties;\n\t\t\t}\n\t\t\t"]))));
    });
    (0, globals_1.it)('should work with any values', function () {
        var theme = {
            x1: {
                name: '--paradigm--x1',
                value: 'val(--paradigm--x1, 4px)',
                originalValue: '4px',
            },
            sizePopupBasePadding: {
                auto: {
                    name: '--vkui--size_popup_base_padding',
                    value: 'var(--vkui--size_popup_base_padding)',
                },
                compact: {
                    name: '--vkui--size_popup_base_padding--compact',
                    value: 'var(--vkui--size_popup_base_padding--compact, 16px)',
                    originalValue: '16px',
                },
                regular: {
                    name: '--vkui--size_popup_base_padding--regular',
                    value: 'var(--vkui--size_popup_base_padding--regular, 20px)',
                    originalValue: '20px',
                },
            },
            colorTextAccent: {
                normal: {
                    name: '--vkui--color_text_accent',
                    value: 'val(--vkui--color_text_accent, white)',
                    originalValue: 'white',
                },
                hover: {
                    name: '--vkui--color_text_accent--hover',
                    value: 'val(--vkui--color_text_accent--hover, black)',
                    originalValue: 'black',
                },
                active: {
                    name: '--vkui--color_text_accent--active',
                    value: 'val(--vkui--color_text_accent--active, red)',
                    originalValue: 'red',
                },
            },
        };
        (0, globals_1.expect)((0, compileGetDeclarationString_1.compileGetDeclarationString)(theme)).toBe((0, common_tags_1.stripIndent)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarString(): string {\n\t\t\t\treturn `--paradigm--x1: 4px;--vkui--size_popup_base_padding--compact: 16px;--vkui--size_popup_base_padding--regular: 20px;--vkui--color_text_accent: white;--vkui--color_text_accent--hover: black;--vkui--color_text_accent--active: red;`;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarObject(): Properties {\n\t\t\t\treturn {\"--paradigm--x1\":\"4px\",\"--vkui--size_popup_base_padding--compact\":\"16px\",\"--vkui--size_popup_base_padding--regular\":\"20px\",\"--vkui--color_text_accent\":\"white\",\"--vkui--color_text_accent--hover\":\"black\",\"--vkui--color_text_accent--active\":\"red\"} as Properties;\n\t\t\t}\n\t\t\t"], ["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarString(): string {\n\t\t\t\treturn \\`--paradigm--x1: 4px;--vkui--size_popup_base_padding--compact: 16px;--vkui--size_popup_base_padding--regular: 20px;--vkui--color_text_accent: white;--vkui--color_text_accent--hover: black;--vkui--color_text_accent--active: red;\\`;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarObject(): Properties {\n\t\t\t\treturn {\"--paradigm--x1\":\"4px\",\"--vkui--size_popup_base_padding--compact\":\"16px\",\"--vkui--size_popup_base_padding--regular\":\"20px\",\"--vkui--color_text_accent\":\"white\",\"--vkui--color_text_accent--hover\":\"black\",\"--vkui--color_text_accent--active\":\"red\"} as Properties;\n\t\t\t}\n\t\t\t"]))));
    });
    (0, globals_1.it)('should work onlyColorsMode', function () {
        var theme = {
            x1: {
                name: '--paradigm--x1',
                value: 'val(--paradigm--x1, 4px)',
                originalValue: '4px',
            },
            sizePopupBasePadding: {
                auto: {
                    name: '--vkui--size_popup_base_padding',
                    value: 'var(--vkui--size_popup_base_padding)',
                },
                compact: {
                    name: '--vkui--size_popup_base_padding--compact',
                    value: 'var(--vkui--size_popup_base_padding--compact, 16px)',
                    originalValue: '16px',
                },
                regular: {
                    name: '--vkui--size_popup_base_padding--regular',
                    value: 'var(--vkui--size_popup_base_padding--regular, 20px)',
                    originalValue: '20px',
                },
            },
            colorTextAccent: {
                normal: {
                    name: '--vkui--color_text_accent',
                    value: 'val(--vkui--color_text_accent, white)',
                    originalValue: 'white',
                },
                hover: {
                    name: '--vkui--color_text_accent--hover',
                    value: 'val(--vkui--color_text_accent--hover, black)',
                    originalValue: 'black',
                },
                active: {
                    name: '--vkui--color_text_accent--active',
                    value: 'val(--vkui--color_text_accent--active, red)',
                    originalValue: 'red',
                },
            },
        };
        (0, globals_1.expect)((0, compileGetDeclarationString_1.compileGetDeclarationString)(theme, 'onlyColors')).toBe((0, common_tags_1.stripIndent)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarColorsString(): string {\n\t\t\t\treturn `--vkui--color_text_accent: white;--vkui--color_text_accent--hover: black;--vkui--color_text_accent--active: red;`;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarColorsObject(): Properties {\n\t\t\t\treturn {\"--vkui--color_text_accent\":\"white\",\"--vkui--color_text_accent--hover\":\"black\",\"--vkui--color_text_accent--active\":\"red\"} as Properties;\n\t\t\t}\n\t\t\t"], ["\n\t\t\timport {Properties} from 'csstype';\n\n\t\t\texport function getDeclarationVarColorsString(): string {\n\t\t\t\treturn \\`--vkui--color_text_accent: white;--vkui--color_text_accent--hover: black;--vkui--color_text_accent--active: red;\\`;\n\t\t\t}\n\n\t\t\texport function getDeclarationVarColorsObject(): Properties {\n\t\t\t\treturn {\"--vkui--color_text_accent\":\"white\",\"--vkui--color_text_accent--hover\":\"black\",\"--vkui--color_text_accent--active\":\"red\"} as Properties;\n\t\t\t}\n\t\t\t"]))));
    });
});
var templateObject_1, templateObject_2, templateObject_3;
