// @vkontakte/vkui-tokens/build/expandTheme.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.expandAll = exports.expandRootTheme = void 0;
var createPseudoRootFromCssVars_1 = require("./themeProcessors/createPseudoRootFromCssVars/createPseudoRootFromCssVars");
var customMedia_1 = require("./themeProcessors/customMedia/customMedia");
var expandColors_1 = require("./themeProcessors/expandColors/expandColors");
var extractCssVarsStrict_1 = require("./themeProcessors/extractCssVarsStrict/extractCssVarsStrict");
var extractGeneralTokens_1 = require("./themeProcessors/extractGeneralTokens/extractGeneralTokens");
var extractVarsNames_1 = require("./themeProcessors/extractVarsNames/extractVarsNames");
var pixelifyValues_1 = require("./themeProcessors/pixelifyValues/pixelifyValues");
/**
 * На основе описания темы, генерирует рутовую тему
 */
function expandRootTheme(themeDescription) {
    var expandedColors = (0, expandColors_1.getExpandedThemeColors)(themeDescription);
    var generalTokens = (0, extractGeneralTokens_1.extractGeneralTokens)(themeDescription);
    var customMedia = (0, customMedia_1.processCustomMedia)(themeDescription);
    return __assign(__assign(__assign(__assign({}, generalTokens), expandedColors), customMedia), { themeType: 'root' });
}
exports.expandRootTheme = expandRootTheme;
/**
 * Генерирует все виды ts-тем, на основе описания темы
 * Может работать и с любым интерфейсом и количеством переменных в теме
 */
function expandAll(themeDescription) {
    var theme = expandRootTheme(themeDescription);
    var pixelifyTheme = (0, pixelifyValues_1.pixelifyValues)(theme);
    var cssVarsThemeWide = (0, extractVarsNames_1.extractVarsNames)(theme);
    var cssVarsTheme = (0, extractCssVarsStrict_1.extractCssVarsStrict)(cssVarsThemeWide);
    var pseudoThemeCssVars = (0, createPseudoRootFromCssVars_1.createPseudoRootFromCssVars)(theme, cssVarsThemeWide);
    return {
        theme: theme,
        pixelifyTheme: pixelifyTheme,
        cssVarsTheme: cssVarsTheme,
        cssVarsThemeWide: cssVarsThemeWide,
        pseudoThemeCssVars: pseudoThemeCssVars,
    };
}
exports.expandAll = expandAll;
