// @vkontakte/vkui-tokens/interfaces/general/tools/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.adaptiveKeys = void 0;
exports.adaptiveKeys = [
    'regular',
    'compactX',
    'compact',
    'large',
    'largeX',
    'largeXX',
];
/*
 * Ниже приведен пример брейкпоинтов и правил перехода из одного в другой для адаптивных переменных
 * Это используется в основном в paradigm темах (исторически) и не является частью основной темы
 * Однако если необходимо получить тему в которой переменные будут переходить автоматически между состояниями
 * а также разные другие вещи такие как правила для media query имеет смысл указать эту секцию
const exampleBreakpoints: Breakpoints<['touch', 'tablet', 'desktopS']> = {
    breakpoints: {
        touch: {
            breakpoint: 0,
            adaptiveValue: 'compact',
        },
        tablet: {
            breakpoint: 768,
            adaptiveValue: 'compact',
        },
        desktopS: {
            breakpoint: 1024,
            adaptiveValue: 'regular',
        },
    },
};
*/
